core/src/main/scala/org/apache/spark/serializer/SerializerHelper.scala

```diff
@@ -0,0 +1,45 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.serializer
+
+import java.nio.ByteBuffer
+
+import scala.reflect.ClassTag
+
+import org.apache.spark.internal.Logging
+import org.apache.spark.util.io.{ChunkedByteBuffer, ChunkedByteBufferOutputStream}
+
+private[spark] object SerializerHelper extends Logging {
+  def serializeToChunkedBuffer[T: ClassTag](
+      serializerInstance: SerializerInstance,
+      t: T): ChunkedByteBuffer = {
+    val cbbos = new ChunkedByteBufferOutputStream(1024 * 1024, ByteBuffer.allocate)
```

Comment:
The default size here is fairly large - most results are fairly small (except for a small subset).
Unfortunately, small `chunkSize` leads to suboptimal performance for `ChunkedByteBufferOutputStream` - this change has a potential for causing performance regressions - something to watch out for.

Reply:
Actually no specific reason for choosing this value, just search around `new ChunkedByteBufferOutputStream` usage, found some `1024 * 1024` in serialize manager and map output tracker, so thought it might be a reasonable value to use.  Smaller size would be good for small task result, while larger chunk size is good at reducing memory fragment in case of large result but a waste for small result.

Also see my comment [below](https://github.com/apache/spark/pull/38064/#issuecomment-1276518707)(just added) about some thoughts on performance